invisible_cities/cities/detsim_test.py

```diff
@@ -0,0 +1,277 @@
+    physics_params["transverse_diffusion"]   = 1.20 * units.mm / units.cm**0.5
+    physics_params["longitudinal_diffusion"] = 0.30 * units.mm / units.cm**0.5
+    physics_params["el_gain"]                = 365
+    physics_params["el_drift_velocity"]      = 2.5  * units.mm / units.mus
+
+    buffer_params = dict()
+    buffer_params["length"]      = 800 * units.mus
+    buffer_params["pmt_width"]   = 100 * units.ns
+    buffer_params["sipm_width"]  =   1 * units.mus
+    buffer_params["max_time"]    =  10 * units.ms
+    buffer_params["pre_trigger"] = 100 * units.mus
+    buffer_params["trigger_thr"] = 0
+
+    # copy to avoid mutability
+    physics_params_ = physics_params.copy()
+    buffer_params_  = buffer_params .copy()
+
+    # modify test file
+    modified_testfile = os.path.join(output_tmpdir, "modified_testfile.h5")
+    shutil.copyfile(PATH_IN, modified_testfile)
+    with tb.open_file(modified_testfile, "a") as h5file:
+        hits = h5file.root.MC.hits
+        readhits = hits.read()
+        hits.remove_rows(start=0)
+        readhits["time"] += 10 * units.mus
+        hits.append(readhits)
+        hits.flush()
+
+    # run over test file
+    conf = configure('detsim $ICTDIR/invisible_cities/config/detsim.conf'.split())
+    conf.update(dict(files_in      = PATH_IN,
+                     file_out      = PATH_OUT,
+                     run_number    = 0,
+                     event_range   = all_events,
+                     physics_params= physics_params,
+                     buffer_params = buffer_params))
+    np.random.seed(1234)
+    result = detsim(**conf)
+
+    # run over modified file
+    modified_file_out = os.path.join(output_tmpdir, "detsim_test_modified.h5")
+    conf = configure('detsim $ICTDIR/invisible_cities/config/detsim.conf'.split())
+    conf.update(dict(files_in      = modified_testfile,
+                     file_out      = modified_file_out,
+                     run_number    = 0,
+                     event_range   = all_events,
+                     physics_params= physics_params_,
+                     buffer_params = buffer_params_))
+    np.random.seed(1234)
+    result = detsim(**conf)
+
+    tables = ["pmtrd", "sipmrd",
+              "Run/eventMap", "Run/events", "Run/runInfo",
+              "MC/configuration", "MC/event_mapping", #"MC/hits", note that MC/hits are different
+              "MC/particles", "MC/sns_positions", "MC/sns_response",
+              "Filters/active_hits"]
+
+    with tb.open_file(PATH_OUT) as output_file:
+        with tb.open_file(modified_file_out) as modified_output_file:
+            for table in tables:
+                assert hasattr(output_file.root, table)
+                got      = getattr( modified_output_file.root, table)
+                expected = getattr(          output_file.root, table)
+                assert_tables_equality(got, expected)
+
+
+def test_detsim_buffer_times(ICDATADIR, output_tmpdir):
```

Comment:
Add a comment on what this test does, for example the nonphysical `el_drift_velocity`is confusing. Also, again,  you can copy dictionaries from conf and update the changed parameters